### PR TITLE
RIA-9583 no_remission_path

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link
https://tools.hmcts.net/jira/browse/RIA-9583

### Change description
Edited MarkPaymentPaidPreparer and MarkPaymentPaidPreparerTest for NO_REMISSION path. As well refactored the classes.

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/MarkPaymentPaidPreparerTest.java [hunk 1]
```diff
@@ -157,18 +154,16 @@ void should_return_error_for_old_pa_ea_hu_cases(AppealType appealType) {
         assertThat(returnedCallbackResponse.getErrors()).contains("You cannot mark this appeal as paid");
     }
 
-    @ParameterizedTest
-    @EnumSource(value = AppealType.class, names = { "PA", "EA", "HU", "EU" })
-    void should_return_error_for_old_pa_ea_hu_cases_when_is_lr_journey_and_dlrm_fee_remission_is_enabled(AppealType appealType) {
+    @Test
+    void should_return_error_for_old_pa_ea_hu_cases_when_is_lr_journey_and_dlrm_fee_remission_is_enabled() {
```
Comment (MiguelMartinRojas) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/MarkPaymentPaidPreparerTest.java:158]: Why did you remove the parametized test?
I case now this condition applies only to PA can you adapt the name of the test?

Comment (ikirsanov) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/MarkPaymentPaidPreparerTest.java:158]: Because it doesnt check anything except PA case on one of conditions because in the old condition (isEaHuEuAg && eaHuPaymentType.isEmpty()   **eaHuPaymentType** is not exists it is not populated anywhere. So we discussed with Rajesh to remove it.

**Review by MiguelMartinRojas — APPROVED** [review]

**Review by ThomasKKC — APPROVED** [review]

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM
